destination/schema/snowflake/datatype.go

```diff
@@ -0,0 +1,370 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snowflake
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/lovromazgon/jsonpoly"
```

Comment:
Note that I extracted the code that handled polymorphic JSON types into a library, as I thought it could be generally useful ([github.com/lovromazgon/jsonpoly](https://github.com/lovromazgon/jsonpoly). In commit https://github.com/conduitio-labs/conduit-connector-snowflake/pull/120/commits/2c8ff922e2a8b1941beea75009d87ea0179f182f I replaced the code in the connector in favor of using the lib.